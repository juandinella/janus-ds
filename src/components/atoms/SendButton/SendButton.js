import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './SendButton.styles'
import ScreenReaderText from '../ScreenReaderText'

/**
 * SendButton
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledSendButton = styled.button.withConfig({ shouldForwardProp })`
  ${styles}
`

const SendButton = ({
  id,
  disabled = false,
  color = '#171717',
  onClick = () => null,
}) => {
  return (
    <StyledSendButton
      id={id}
      data-testid={id}
      onClick={onClick || null}
      disabled={disabled}
    >
      <svg
        width="24"
        height="24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M22.707 1.293a1 1 0 0 1 0 1.414l-11 11a1 1 0 0 1-1.414-1.414l11-11a1 1 0 0 1 1.414 0Z"
          fill={color}
        />
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M22.707 1.293a1 1 0 0 1 .237 1.037l-7 20a1 1 0 0 1-1.858.076l-3.844-8.648-8.648-3.844a1 1 0 0 1 .076-1.858l20-7a1 1 0 0 1 1.037.237ZM4.711 9.11l6.695 2.975a1 1 0 0 1 .508.508l2.975 6.695L20.37 3.63 4.711 9.11Z"
          fill={color}
        />
      </svg>
      <ScreenReaderText>Send Message</ScreenReaderText>
    </StyledSendButton>
  )
}

SendButton.displayName = 'SendButton'

SendButton.propTypes = {
  id: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  color: PropTypes.string,
}

export default SendButton
