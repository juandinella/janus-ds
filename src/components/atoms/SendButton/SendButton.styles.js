import { css } from 'styled-components'

export default css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 42px;
  height: 42px;
  min-width: 42px;
  min-height: 42px;
  max-width: 42px;
  max-height: 42px;
  background-color: var(--bg-base);
  border-radius: 100%;
  transition: background-color var(--transition-225);

  &:hover {
    background-color: var(--bg-primary-lightest);
  }

  &:disabled {
    cursor: default;
    opacity: 0.5;
  }
`
