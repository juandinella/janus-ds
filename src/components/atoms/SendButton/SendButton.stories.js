import SendButton from './SendButton'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(SendButton)

export default {
  title: 'Atoms/SendButton',
  component: SendButton,
  args: {
    color: '#171717',
  },
  argTypes: {
    color: {
      control: { type: 'color' },
    },
  },
}

export const Default = Template.bind({})

export const Disabled = { args: { disabled: true } }
