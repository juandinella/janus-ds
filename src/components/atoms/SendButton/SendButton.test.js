import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import React from 'react'
import SendButton from './SendButton'

initStoryshots({
  storyKindRegex: /^Atoms\/SendButton$/,
})

const componentProps = {
  id: 'unique_id',
  onClick: jest.fn(),
}
afterEach(cleanup)
beforeEach(() => {
  componentProps.onClick.mockClear()
})

describe('<SendButton />', () => {
  test('should render component correctly', () => {
    render(<SendButton {...componentProps} />)
    const btn = screen.getByTestId(componentProps.id)
    expect(btn).toBeInTheDocument()
  })
  test('should call function on click', () => {
    const { getByTestId } = render(<SendButton {...componentProps} />)
    const button = getByTestId(componentProps.id)
    fireEvent.click(button)
    fireEvent.click(button)
    expect(componentProps.onClick).toHaveBeenCalledTimes(2)
  })
})
