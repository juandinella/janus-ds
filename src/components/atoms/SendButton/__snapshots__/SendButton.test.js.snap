// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/SendButton Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <button
    className="sc-1x831cm-0 dZNuRb"
    disabled={false}
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M22.707 1.293a1 1 0 0 1 0 1.414l-11 11a1 1 0 0 1-1.414-1.414l11-11a1 1 0 0 1 1.414 0Z"
        fill="#171717"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M22.707 1.293a1 1 0 0 1 .237 1.037l-7 20a1 1 0 0 1-1.858.076l-3.844-8.648-8.648-3.844a1 1 0 0 1 .076-1.858l20-7a1 1 0 0 1 1.037.237ZM4.711 9.11l6.695 2.975a1 1 0 0 1 .508.508l2.975 6.695L20.37 3.63 4.711 9.11Z"
        fill="#171717"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="sc-90c4yq-0 bmmsAZ"
    >
      Send Message
    </span>
  </button>
</div>
`;

exports[`Storyshots Atoms/SendButton Disabled 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <button
    className="sc-1x831cm-0 dZNuRb"
    disabled={true}
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M22.707 1.293a1 1 0 0 1 0 1.414l-11 11a1 1 0 0 1-1.414-1.414l11-11a1 1 0 0 1 1.414 0Z"
        fill="#171717"
        fillRule="evenodd"
      />
      <path
        clipRule="evenodd"
        d="M22.707 1.293a1 1 0 0 1 .237 1.037l-7 20a1 1 0 0 1-1.858.076l-3.844-8.648-8.648-3.844a1 1 0 0 1 .076-1.858l20-7a1 1 0 0 1 1.037.237ZM4.711 9.11l6.695 2.975a1 1 0 0 1 .508.508l2.975 6.695L20.37 3.63 4.711 9.11Z"
        fill="#171717"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="sc-90c4yq-0 bmmsAZ"
    >
      Send Message
    </span>
  </button>
</div>
`;
